src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -49,128 +45,141 @@ public HttpConnectionDispatcher(ConnectionManager manager, ILoggerFactory logger
 
         private async Task ExecuteEndpointAsync(string path, HttpContext context, EndPoint endpoint)
         {
-            // Server sent events transport
-            if (context.Request.Path.StartsWithSegments(path + "/sse"))
```

Comment:
The biggest problem is that there's no consistent transport-agnostic way to look up the connection because WebSockets allows lazy initialization of the connection. That's why refactoring to reduce duplication didn't really reduce complexity. It certainly didn't increase it based on my measurement.